src/index.ts

```diff
@@ -5,12 +5,18 @@ export enum managerType {
     pip = "pip",
     gem = "gem",
     composer = "composer"
-
 }
 
 
-export function getManager(type: managerType): Manager {
-    const managerName: string = managerType[type];
-    const managerClass = require(`./managers/${managerName}`);
-    return new managerClass.default(managerName);
-}
+export function getManager(manager: managerType | String): Manager | undefined {
+    const managerName: string | undefined =
+        manager instanceof (String) ?
+            ((Object as any).values(managerType).includes(manager.toString()) ? manager.toString() : undefined) :
+            managerType[manager];
+    if (managerName){
+        const managerClass = require(`./managers/${managerName}`);
```

Comment:
Instead of requesting/including js file y name I strongly recommend to define them explicitly.

Reply:
+ if the enum is defined why do you overcomplicate the logic that checks the string (manager)

Reply:
And it's actually registry...